app/src/main/java/com/android/streetworkapp/ui/navigation/TopAppBarManager.kt

```diff
@@ -0,0 +1,35 @@
+package com.android.streetworkapp.ui.navigation
+
+import androidx.compose.material.icons.Icons
+import androidx.compose.material.icons.automirrored.filled.ArrowBack
+import androidx.compose.ui.graphics.vector.ImageVector
+
+class TopAppBarManager(
+    private var title: String = "",
+    private var hasNavigationIcon: Boolean = false,
+    private var navigationIcon: ImageVector? = null
+) {
+
+  companion object {
+    val DEFAULT_TOP_APP_BAR_NAVIGATION_ICON = Icons.AutoMirrored.Filled.ArrowBack
```

Comment:
Suggestion: change the `val` into `var` this improves immutability and reduces potential side effects.

Reply:
I'm not sure I understand this, changing the val into var would be worse for immutability

Reply:
Sorry I was indeed confused, my bad resolved